v19.2/tutorials/multi-region-setup.md

```diff
@@ -0,0 +1,157 @@
+    ~~~
+    ~~~
+      table_name
+    +------------+
+      rides
+      users
+      vehicles
+    (3 rows)
+    ~~~
+
+
+## Setting up a virtual development environment
+
+In production, you want to [containerize](https://www.docker.com/resources/what-container) your application and deploy it with a deployment orchestration tool, like [Kubernetes](https://kubernetes.io/). For debugging, use [`pipenv`](https://docs.pipenv.org/en/latest/install/#installing-pipenv), a tool that manages dependencies with `pip` and creates virtual environments with `virtualenv`.
+
+1. Run the following command to initialize the project's virtual environment:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ pipenv --three
+    ~~~
+
+    `pipenv` creates a `Pipfile` in the current directory. Open this `Pipfile`, and confirm its contents match the following:
+
+
+    ~~~ toml
+    [[source]]
+    name = "pypi"
+    url = "https://pypi.org/simple"
+    verify_ssl = true
+
+    [dev-packages]
+
+    [packages]
+    cockroachdb = "*"
+    psycopg2-binary = "*"
+    SQLAlchemy = "*"
+    SQLAlchemy-Utils = "*"
+    Flask = "*"
+    Flask-SQLAlchemy = "*"
+    Flask-WTF = "*"
+    Flask-Bootstrap = "*"
+    Flask-Login = "*"
+    WTForms = "*"
+    gunicorn = "*"
+    geopy = "*"
+
+    [requires]
+    python_version = "3.7"
+    ~~~
+
+1. Run the following command to install the packages listed in the `Pipfile`:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ pipenv install
+    ~~~
+
+1. To connect to a SQL database (like CockroachDB) from a client, you need a [SQL connection string](https://en.wikipedia.org/wiki/Connection_string). Rather than hard-coding the connection string into the source code, the application reads it from an environment variable. Pipenv automatically sets any variables defined in a `.env` file as environment variables in a Pipenv virtual environment.
+
+    So, open `.env`, and edit the `DB_URI` environment variable so that it matches the connection string for the demo cluster that you started earlier (you should just need to change the `<port>`). Note that SQLAlchemy requires the connection string protocol to be specific to the CockroachDB dialect.
+
+    For example:
+
+    ~~~
+    DB_URI = 'cockroachdb://root@127.0.0.1:52382/movr'
+    ~~~
+
+    You can also specify other variables in this file that you'd rather not hard-code in the application, like API keys and secret keys used by the application. For debugging purposes, you should leave these variables as they are.
+
+
+1. Activate the virtual environment:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ pipenv shell
+    ~~~
+
+    The prompt should now read `~bash-3.2$`. From this shell, you can run any Python 3 application with the required dependencies that you listed in the `Pipfile`, and the environment variables that you listed in the `.env` file. You can exit the shell subprocess at any time with a simple `exit` command.
```

Comment:
This exact prompt string probably only applies on macOS, which uses an ancient pre-GPLv3 `bash` for legal reasons.  Also, not everyone uses bash (I don't).  Perhaps this could say something general like "the prompt should change to your shell's default prompt.  This signals that you are now in the Python virtual environment"?